routing/router.go

```diff
@@ -2311,6 +2311,17 @@ func (r *ChannelRouter) BuildRoute(amt *lnwire.MilliSatoshi,
 		return nil, err
 	}
 
+	routingTx, err := newDbRoutingTx(r.cfg.Graph)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := routingTx.close()
```

Comment:
won't this tx be open much longer than necessary now? It actually looks like we'll do an RPC call inside it below...

Reply:
Moved rpc call outside the db tx